### DISCUSSION TITLE
middleware can't get client's cookies when access directly?

### DISCUSSION THREAD

**Discussion by tomohirohiratsuka** [vercel/next.js#39697] [Help] [answered]
I have a middleware which is handling protected routes.
And inside middleware I'm using cookies like `request.headers.get('cookie')`

On local, when I access page by Link or directly input url on browser, both of them are working well.
But on production environment `request.headers.get('cookie')` is returning null even though browser has cookies.
Why on production env, I can't get cookies? 
I'd appreciate if anyone give me hints.
Thank you.
```ts:middleware.ts
export async function middleware(request: NextRequest) {
  console.log(request.headers.get('cookie'));
  const response = NextResponse.next();
  // some api calls with await.
  return response;
}
```

My environment is here.
```
+-- next@12.2.3
+-- react@18.2.0
+-- typescript@4.7.4
```

**Comment by tomohirohiratsuka** [comment, marked as answer]
This was my misconfiguration.
Solved.

**Reply by inahsanjuan** [reply]
How did you solve it btw? What config did you mistakenly put?

**Reply by tomohirohiratsuka** [reply]
Actually this was not related to nextjs, it was env variables misconfiguration.
I put wrong domain mistakenly on .env.
